lib/mayaUsd/ufe/UsdUndoCreateGroupCommand.cpp

```diff
@@ -81,10 +81,10 @@ void UsdUndoCreateGroupCommand::execute()
 
     auto newParentHierarchy = Ufe::Hierarchy::hierarchy(_group);
     if (newParentHierarchy) {
-        for (auto child : _selection) {
+        for (const auto& child : _selection) {
             auto parentCmd = newParentHierarchy->appendChildCmd(child);
-            parentCmd->execute();
             append(parentCmd);
+            parentCmd->execute();
```

Comment:
@ppt-adsk this logics seems a bit awkward.  I believe the order matters here. We want to append the commands to the CmdList before running the execute().